tests/test_ocr_runner_unit.py

```
@@ -0,0 +1,72 @@
+    out_pdf = tmp_path / 'out.pdf'
+    log = tmp_path / 'log.txt'
+    input_pdf.write_bytes(b'%PDF-1.4')
+    rc = ocr_runner.run_ocrmypdf(str(input_pdf), str(out_pdf), str(log), timeout=1)
+    assert rc == 0
+
+
+def test_fallback_tesseract(monkeypatch, tmp_path):
+    # emulate pdftoppm creating PNG files and tesseract returning text
+    created = []
+
+    def fake_pdftoppm(args, stdout=None, stderr=None, timeout=None):
```

Ruff (0.14.10)

Missing return type annotation for private function `fake_pdftoppm`

(ANN202)